extension/lib/bigcommerce/RequestRepository.js

```diff
@@ -0,0 +1,92 @@
+const BigCommerceLogger = require('./Logger')
+
+class BigCommerceRequestRepository {
+  /**
+   * @param {BigCommerce} client Api V3 client
+   * @param {Logger} logger context.log
+   */
+  constructor (client, logger) {
+    this.client = client
+    this.logger = logger
+  }
+
+  /**
+   * @param {string} path
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  get (path) {
+    return this.request('get', path)
+  }
+
+  /**
+   * @param {string} path
+   * @param {Object} data
+   * @param {Object} [obfuscatedData]
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  post (path, data, obfuscatedData = null) {
+    return this.request('post', path, data, obfuscatedData)
+  }
+
+  /**
+   * @param {string} path
+   * @param {Object} data
+   * @param {Object} [obfuscatedData]
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  put (path, data, obfuscatedData = null) {
+    return this.request('put', path, data, obfuscatedData)
+  }
+
+  /**
+   * @param {string} path
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  del (path) {
+    return this.request('delete', path)
+  }
+
+  /**
+   * @param {string} type
+   * @param {string} path
+   * @param {Object} [data]
+   * @param {Object} [obfuscatedData]
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  async request (type, path, data = null, obfuscatedData = null) {
+    const requestData = this.getRequestData(type, path, data, obfuscatedData)
```

Comment:
the naming is a bit off, this request data is possibly obfuscated data, so the naming might not make sense. I would expect you to use requestData and pass it to the request at this point. Maybe it would be called `obfuscatedData`?

Reply:
changed it to logData because obfuscatedData is a param and that could be confusing